### PR TITLE
Zetia/fix ssl secret flag

### PR DESCRIPTION
**Pull request by zetiaatgithub**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [ ] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [x] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your pull request is merged into main branch, a new pull request will be created to update `src/index.json` automatically.  
The precondition is to put your code inside this repository and upgrade the version in the pull request but do not modify `src/index.json`.

### CODE REVIEW

File: src/k8s-extension/azext_k8s_extension/partner_extensions/AzureMLKubernetes.py [hunk 1]
```diff
@@ -366,14 +366,17 @@ def Update(self, cmd, resource_group_name, cluster_name, auto_upgrade_minor_vers
...
-
-                if fe_ssl_cert_file and fe_ssl_key_file:
-                    self.__set_inference_ssl_from_file(configuration_protected_settings, fe_ssl_cert_file, fe_ssl_key_file)
+        fe_ssl_secret = _get_value_from_config_protected_config(
+            self.SSL_SECRET, configuration_settings, configuration_protected_settings)
+        fe_ssl_cert_file = configuration_protected_settings.get(self.sslCertPemFile)
+        fe_ssl_key_file = configuration_protected_settings.get(self.sslKeyPemFile)
+        # always take ssl key/cert first, then secret if key/cert file is not provided
+        if fe_ssl_cert_file and fe_ssl_key_file:
+            logger.info(f"Both {self.sslKeyPemFile} and {self.sslCertPemFile} are set, update ssl key.")
```
Comment (NarayanThiru) [inline src/k8s-extension/azext_k8s_extension/partner_extensions/AzureMLKubernetes.py:375]: Minor, reword the message to "...set, **updating** ssl key."

File: src/k8s-extension/azext_k8s_extension/partner_extensions/AzureMLKubernetes.py [hunk 2]
```diff
@@ -366,14 +366,17 @@ def Update(self, cmd, resource_group_name, cluster_name, auto_upgrade_minor_vers
...
+            self.__set_inference_ssl_from_file(configuration_protected_settings, fe_ssl_cert_file, fe_ssl_key_file)
+        elif fe_ssl_secret:
+            logger.info(f"{self.SSL_SECRET} is set, update ssl secret.")
```
Comment (NarayanThiru) [inline src/k8s-extension/azext_k8s_extension/partner_extensions/AzureMLKubernetes.py:378]: Here too reword the message to "...set, **updating** ssl secret."

**Review by NarayanThiru — CHANGES_REQUESTED** [review]

File: src/k8s-extension/azext_k8s_extension/partner_extensions/AzureMLKubernetes.py (hunk 1 above)
Comment (zetiaatgithub) [reply src/k8s-extension/azext_k8s_extension/partner_extensions/AzureMLKubernetes.py]: Fixed

File: src/k8s-extension/azext_k8s_extension/partner_extensions/AzureMLKubernetes.py (hunk 2 above)
Comment (zetiaatgithub) [reply src/k8s-extension/azext_k8s_extension/partner_extensions/AzureMLKubernetes.py]: Fixed

**Review by NarayanThiru — APPROVED** [review]

**Comment by NarayanThiru** [comment]
@zetiaatgithub , can you investigate the test failure and fix it.  It's one of the AzureML tests that is failing

**Pull request merged**